mlflow/models/dependencies_schemas.py

```diff
@@ -56,11 +56,17 @@ def set_retriever_schema(
                 other_columns=["title"],
             )
     """
-    globals()[_RETRIEVER_PRIMARY_KEY] = primary_key
-    globals()[_RETRIEVER_TEXT_COLUMN] = text_column
-    globals()[_RETRIEVER_DOC_URI] = doc_uri
-    globals()[_RETRIEVER_OTHER_COLUMNS] = other_columns or []
-    globals()[_RETRIEVER_NAME] = name
+    retriever_schema = globals().get(DependenciesSchemasType.RETRIEVERS.value, [])
```

Comment:
The reason why its a list, is so that I didn't want to change the API how its consumed in the trace + in the MLModel file this is added as a list.
But this is a good point. `same retriever name correspond to different schemas`. We could throw an error if we see that. Let me add that to this PR :)

Reply:
I think this is fine for now if this is the only user-facing API.
But I think as a follow-up we should update this to dictionary, and also make APIs consistent:
Set schema on a retriever `set_retriever_schema(name=..., schema=...)`
Retrieve schema on a retriever `get_retriever_schema(name=...)`
Remove schema on a retriever `remove_retrieve_schema(name=...)`

Now it's like that we call `set_retriever_schema` but actually it's `appending` schemas. And `get`/`clear` is dealing with the list of schemas.

Reply:
Yes, but users never ideally need to call `get_retriever_schema` or ` remove_retrieve_schema`. Its an internal API. So users just call `set_retriever_schema` which I didn't want to change. Ideally it is experimental so I could break it. But not sure if we want to do now. Maybe if we feel this is not scaling, we could move this to a dict